src/agent/rustjail/src/mount.rs

```diff
@@ -1041,7 +1044,10 @@ fn mask_path(path: &str) -> Result<()> {
 
 fn readonly_path(path: &str) -> Result<()> {
     if !path.starts_with('/') || path.contains("..") {
-        return Err(anyhow!(nix::Error::EINVAL));
+        return Err(anyhow!(
```

Comment:
It might be worth creating a common check function for this and `mask_path()` to avoid the duplicated code so both functions could start with something like:

```rust
check_path(path)?;

// ...
```

Reply:
... if you did do this (possibly on anther PR), you could consider writing a set of table-based tests:

- https://github.com/kata-containers/kata-containers/blob/main/docs/Unit-Test-Advice.md#rust-table-driven-tests

There are lots of examples of these in the repo - search for `TestData` to find a bunch.